From 44934f66441d4835158c58c28cebaa15ada3bc63 Mon Sep 17 00:00:00 2001
From: Shreenidhi Shedi <sshedi@vmware.com>
Date: Tue, 22 Mar 2022 20:30:32 +0530
Subject: [PATCH 20/39] allow lvm_t to transit to unconfined_t

This fixes the below issue while running dosfstool & thin-provisioning-tool tests.

[   73.542614] audit: type=1400 audit(1647960973.073:47): avc:  denied  { associate } for  pid=745 comm="dmsetup" key=223178746  scontext=system_u:system_r:lvm_t:s0-s0:c0.c1023 tcontext=system_u:system_r:unconfined_t:s0-s0:c0.c1023 tclass=sem permissive=1
[   73.542781] audit: type=1400 audit(1647960973.073:48): avc:  denied  { unix_read } for  pid=745 comm="dmsetup" key=223178746  scontext=system_u:system_r:lvm_t:s0-s0:c0.c1023 tcontext=system_u:system_r:unconfined_t:s0-s0:c0.c1023 tclass=sem permissive=1
[   73.542783] audit: type=1400 audit(1647960973.073:48): avc:  denied  { read } for  pid=745 comm="dmsetup" key=223178746  scontext=system_u:system_r:lvm_t:s0-s0:c0.c1023 tcontext=system_u:system_r:unconfined_t:s0-s0:c0.c1023 tclass=sem permissive=1
[   73.542888] audit: type=1400 audit(1647960973.073:49): avc:  denied  { unix_write } for  pid=745 comm="dmsetup" key=223178746  scontext=system_u:system_r:lvm_t:s0-s0:c0.c1023 tcontext=system_u:system_r:unconfined_t:s0-s0:c0.c1023 tclass=sem permissive=1
[   73.542890] audit: type=1400 audit(1647960973.073:49): avc:  denied  { write } for  pid=745 comm="dmsetup" key=223178746  scontext=system_u:system_r:lvm_t:s0-s0:c0.c1023 tcontext=system_u:system_r:unconfined_t:s0-s0:c0.c1023 tclass=sem permissive=1

Signed-off-by: Shreenidhi Shedi <sshedi@vmware.com>
---
 policy/modules/system/lvm.te | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/policy/modules/system/lvm.te b/policy/modules/system/lvm.te
index c9e20b0..63cec66 100644
--- a/policy/modules/system/lvm.te
+++ b/policy/modules/system/lvm.te
@@ -421,3 +421,7 @@ optional_policy(`
 	xen_append_log(lvm_t)
 	xen_dontaudit_rw_unix_stream_sockets(lvm_t)
 ')
+
+optional_policy(`
+    unconfined_domain(lvm_t);
+')
-- 
2.45.2

